core/traffic_class.cc

```diff
@@ -344,6 +344,10 @@ bool LeafTrafficClass::RemoveTask(Task *t) {
   auto it = std::find(tasks_.begin(), tasks_.end(), t);
   if (it != tasks_.end()) {
     tasks_.erase(it);
+    // Avoid out-of-bounds access in RunTasks()
+    if (task_index_) {
```

Comment:
Perhaps there's no need for the if?  Just set task_index_ to 0 whenever a task is removed.